task/tkn-bundle/0.1/tkn-bundle.yaml

```diff
@@ -96,15 +100,42 @@ spec:
           done
         fi
       done
+      for f in "${FILES[@]}"; do
+        if [[ -n "$(params.STEPS_IMAGE)" ]]; then
+          yq e '(.spec.steps[] | select(has("image")).image) = "$(params.STEPS_IMAGE)"' -i $f
+        fi
+        # filter any duplicate entries and write to workspace file
+        [[ ! `grep "${f}" "${TASK_FILE}"` ]] && echo "$f" >> "${TASK_FILE}"
+      done
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/openshift-pipeline/openshift-pipelines-cli-tkn:5.0
+    name: build
+    env:
+    - name: CONTEXT
+      value: $(params.CONTEXT)
+    - name: IMAGE
+      value: $(params.IMAGE)
+    script: |
+      #!/bin/env bash
+
+      set -o errexit
+      set -o pipefail
+      set -o nounset
+
+      mapfile -t FILES < "${TASK_FILE}"
```

Comment:
Reads the file into an array.

Reply:
But why was it not needed previously?

Reply:
Because I split the task into multiple steps. The first step passes information using the workspace to the second step. This was needed because the `tkn-cli` image only contained `sed` and `awk` to manipulate the yaml files and I thought that solution was too messy.